conformance/lang/expressions/list-constructor-expr/spread_operator.balt

```diff
@@ -0,0 +1,728 @@
+Test-Case: output
+Description: Test spread operator with basic contextually expected type.
+Labels: array-type, list-constructor-expr, spread-operator, tuple-type
+
+function init() {
+    int[] a0 = [];
+    int[2] a1 = [];
+    int[2] a2 = [1, 2];
+    int[] v0 = [...a0];
+    [int, int] v1 = [...a1];
+    int[] v2 = [...a2];
+    io:println(v0); // @output []
+    io:println(v1); // @output [0,0]
+    io:println(v2); // @output [1,2]
+    io:println(v2); // @output [1,2]
+}
+
+Test-Case: output
+Description: Test spread operator with record contextually expected type.
+Labels: array-type, list-constructor-expr, spread-operator, tuple-type, mapping-constructor-expr, module-type-defn, record-type
+
+type Foo record {|int i;|};
+
+function init() {
+    Foo[] a0 = [];
+    Foo[2] a1 = [{i:10}, {i:15}];
+    Foo[] v0 = [...a0];
+    [Foo, Foo] v1 = [...a1];
+    Foo[] v2 = [...a1];
+    io:println(v0); // @output []
+    io:println(v1); // @output [{"i":10},{"i":15}]
+    io:println(v2); // @output [{"i":10},{"i":15}]
+}
+
+Test-Case: output
+Description: Test spread operator with union of record types.
+Labels: array-type, list-constructor-expr, spread-operator, tuple-type, mapping-constructor-expr, module-type-defn, record-type,
+        union-type
+
+type Foo record {|int i;|};
+type Bar string|int|Foo;
+
+function init() {
+    (Bar|Foo)[] a0 = [];
+    (Bar|Foo)[4] a1 = [{i:10}, {i:15}, 1, 2];
+    (Bar|Foo)[] v0 = [...a0];
+    [Bar|Foo, Bar|Foo, Bar|Foo, Bar|Foo] v1 = [...a1];
+    (Bar|Foo)[] v2 = [...a1];
+    io:println(v0); // @output []
+    io:println(v1); // @output [{"i":10},{"i":15},1,2]
+    io:println(v2); // @output [{"i":10},{"i":15},1,2]
+}
+
+Test-Case: output
+Description: Test multiple spread operators with list constructor expr.
+Labels: array-type, list-constructor-expr, spread-operator, var
+
+function init() {
+    int[] a0 = [];
+    int[2] a1 = [];
+    int[2] a2 = [1, 2];
+    int[] a3 = [2, 2, 2];
+    int[] v0 = [...a0, ...a1];
+    var v1 = [...a1, ...a2];
+    int[] v2 = [...a2, ...a1];
+    int[] v3 = [...a0, ...a2, ...a1];
+    int[] v4 = [...a0, ...a2, ...a1, ...a3];
+    io:println(v0); // @output [0,0]
+    io:println(v1); // @output [0,0,1,2]
+    io:println(v2); // @output [1,2,0,0]
+    io:println(v3); // @output [1,2,0,0]
+    io:println(v4); // @output [1,2,0,0,2,2,2]
+}
+
+Test-Case: output
+Description: Test spread operator when there is union contextually expected type.
```

Comment:
Here the contextually expected type of the spread operator is not a list of union right?
```suggestion
Description: Test spread operator when there is a list of union typed members as the contextually expected type for the list constructor expression.
```